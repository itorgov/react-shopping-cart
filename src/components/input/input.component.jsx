import React from 'react';

import styles from './input.module.scss';

export const Input = ({id, type, value, placeholder, label, onChange}) => (
    <div className={styles.group}>
        {label ? <label htmlFor={id} className={styles.label}>{label}</label> : ''}
        <input
            id={id}
            type={type}
            value={value}
            placeholder={placeholder}
            className={styles.input}
            onChange={onChange}
        />
    </div>
);
