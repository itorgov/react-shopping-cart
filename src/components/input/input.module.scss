.group {
  display: flex;
  flex-direction: column;
}

.label {
  text-transform: uppercase;
  letter-spacing: 1px;
  font-size: 1.2rem;
  font-weight: 600;
  color: var(--color-text-light);
  margin-bottom: 1rem;
}

.input {
  font-family: inherit;
  font-size: 1.6rem;
  color: var(--color-text-light);
  padding: 1rem 1.5rem;
  border: 1px solid rgba(var(--color-black-rgb), 0.1);
  border-radius: 10rem;

  &::placeholder {
    color: var(--color-text-light);
    opacity: 0.8;
  }

  &:focus {
    outline: none;
    box-shadow: 0 1rem 3rem rgba(var(--color-secondary-rgb), 0.1);
  }
}
