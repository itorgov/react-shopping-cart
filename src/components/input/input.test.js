import React from "react";
import {shallow} from "enzyme";

import {Input} from './input.component';

describe('<Input />', () => {
    it('expect to render', () => {
        expect(shallow(<Input />)).toMatchSnapshot();
    });

    it('expect to render with label when it passed as a prop', () => {
        expect(shallow(<Input label='Test label' id='test' />)).toMatchSnapshot();
    });
});
