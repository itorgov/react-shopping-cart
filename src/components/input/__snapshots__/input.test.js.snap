// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> expect to render 1`] = `
<div
  className="group"
>
  <input
    className="input"
  />
</div>
`;

exports[`<Input /> expect to render with label when it passed as a prop 1`] = `
<div
  className="group"
>
  <label
    className="label"
    htmlFor="test"
  >
    Test label
  </label>
  <input
    className="input"
    id="test"
  />
</div>
`;
